lib/Doctrine/ORM/Query/SqlWalker.php

```diff
@@ -1148,7 +1150,10 @@ public function walkJoin($join)
                     $conditions[] = $filterExpr;
                 }
 
-                $sql .= $condExprConjunction . implode(' AND ', $conditions);
+                if ($conditions) {
```

Comment:
No, because you'd add the `$condExprConjunction` to the `$sql` (an `AND` or `ON`) even when no actual condition would follow.